src/platform/interfaces/np_udp.h

```diff
@@ -28,8 +28,24 @@ struct np_udp {
  * This struct defines a list of functions which is required for udp
  * communication.
  *
- * Each function needs to point to a specific platform dependent
- * implementation.
+ * Each function needs to point to a specific platform dependent implementation.
+ *
+ * Error handling. The UDP module expects functions to return or resolve
+ * completion events in the following manners:
+ *
+ * Recoverable errors: If an recoverable errors occurs the functions should
+ * return or resolve with NABTO_EC_OK or NABTO_EC_PACKET_NOT_SENT. Recoverable
+ * errors are all those errors which does not mean that we should give up on the
+ * socket. It could be transient errors such as temporary memory issues and
+ * network connectivity issues. Examples: ENOBUFS, ENOMEM, EAGAIN, EWOULDBLOCK,
+ * etc.
+ *
+ * Fatal errors are all those errors which is not recoverable such as EBADFD
+ * EINVAL, etc.
+ *
+ * Since error codes are not specified for all possible errors on all possible
```

Comment:
we should have this explanation for TCP as well

Reply:
In TCP we do not resolve the functions if recoverable errors occurs, we wait for more data or maybe resolve with a length 0 buffer. For tcp send we always send all the data before resolving the future, if that is not possible it is an unrecoverable error.